### PR TITLE
Remove MSAL from the ADAL repo

### PR DESCRIPTION
**Pull request by MarkZuber**
This won't merge until we complete the MSAL repo validation, but for review this is the code to get us to compile on the ADAL side by removing the MSAL projects, merging the core project into adal, and getting bits to compile.

### CODE REVIEW

File: build/credscan-exclusion.json [hunk 1]
```diff
@@ -44,6 +44,26 @@
     {
       "file": "CryptographyTests.cs",
       "_justification": "Password contained within this file is not used for any production application/code. It is used as a means to access a self signed certificate used only for testing."
-    }
+    },
+{
```
Comment (henrik-me) [inline build/credscan-exclusion.json]: nit: indentation

File: build/credscan-exclusion.json [hunk 2]
```diff
@@ -44,6 +44,26 @@
...
+    {
+      "file": "valid_cert.pfx",
+      "_justification": "Self-signed certificate used by unit tests."
+    },
+    {
+      "file": "valid_cert2.pfx",
+      "_justification": "Self-signed certificate used by unit tests."
+    },
+    {
+      "file": "invalid_cert.pfx",
```
Comment (henrik-me) [inline build/credscan-exclusion.json]: seems like some of these are duplicates. are all needed?

**Comment by henrik-me** [comment]
> Version 4.4.1

This file is also linked to in the wiki I believe. There might be a link to update somewhere.

---
Refers to: changelog.txt:1 in aa7d998. [](commit_id = aa7d9984966dc6980e68e6fb5168b9cdc1f6cdaf, deletion_comment = False)

**Review by henrik-me — APPROVED** [review]
:shipit:

File: build/credscan-exclusion.json (hunk 2 above)
Comment (MarkZuber) [reply build/credscan-exclusion.json]: fixed

File: build/credscan-exclusion.json (hunk 1 above)
Comment (MarkZuber) [reply build/credscan-exclusion.json]: fixed

**Review by bgavrilMS — APPROVED** [review]
LGTM

**Pull request merged**